src/main.rs

```diff
@@ -100,89 +91,96 @@ async fn main_loop(mut io: IOManager, steam_api_requester: Sender<SteamID>) {
     let mut new_players = Vec::new();
 
     loop {
+        // Log
         match io.handle_log() {
-            Ok(Some(output)) => {
+            Ok(output) => {
                 let mut state = State::write_state();
                 state.log_file_state = Ok(());
                 if let Some(new_player) = state.server.handle_io_response(output) {
                     new_players.push(new_player);
                 }
             }
-            Ok(None) => {}
             Err(e) => {
-                State::write_state().log_file_state = Err(e);
+                let mut state = State::write_state();
+                // This one runs very frequently so we'll only print diagnostics
+                // once when it first fails.
+                if state.log_file_state.is_ok() {
+                    tracing::error!("Failed to get log file contents: {:?}", e);
+                }
+                state.log_file_state = Err(e);
             }
         }
 
+        // Commands
         match io.handle_waiting_command().await {
             Ok(output) => {
                 let mut state = State::write_state();
                 state.rcon_state = Ok(());
-                if let Some(output) = output {
-                    if let Some(new_player) = state.server.handle_io_response(output) {
-                        new_players.push(new_player);
-                    }
+                if let Some(new_player) = state.server.handle_io_response(output) {
+                    new_players.push(new_player);
                 }
             }
             Err(e) => {
+                tracing::error!("Failed to run command: {:?}", e);
                 State::write_state().rcon_state = Err(e);
             }
         }
 
         // Request steam API stuff on new players and clear
         for player in &new_players {
-            steam_api_requester.send(*player).await.unwrap();
+            steam_api_requester
+                .send(*player)
+                .await
+                .expect("Steam API task ded");
         }
 
         new_players.clear();
     }
 }
 
 async fn refresh_loop(cmd: Sender<Commands>) {
-    log::debug!("Entering refresh loop");
+    tracing::debug!("Entering refresh loop");
     loop {
         State::write_state().server.refresh();
 
-        cmd.send(Commands::Status).await.unwrap();
+        cmd.send(Commands::Status).await.expect("Command loop ded");
```

Comment:
💀 please better err message uwu

Reply:
Aww, I thought it was funny heh. All g I've  changed the messages.